membership-attribute-service/app/models/AccountDetails.scala

```diff
@@ -78,11 +74,11 @@ object AccountDetails {
       def jsonifyPlan(plan: SubscriptionPlan.AnyPlan) = Json.obj(
         "name" -> externalisePlanName(plan),
         "start" -> plan.start,
+        "end" -> plan.end,
         // if the customer acceptance date is future dated (e.g. 6for6) then always display, otherwise only show if starting less than 30 days from today
         "shouldBeVisible" -> (subscription.acceptanceDate.isAfter(now) || plan.start.isBefore(now.plusDays(30)))
       ) ++ (plan match {
-        case paidPlan: SubscriptionPlan.Paid => Json.obj(
-          "end" -> paidPlan.end,
+        case paidPlan: PaidSubscriptionPlan[_, _] => Json.obj(
```

Comment:
changing this line was to avoid a misleading Type Erasure warning (which came from using the type alias `SubscriptionPlan.Paid` which references type params which do get erased but aren't needed for the pattern match